osf/migrations/0142_add_spam_fields_to_user.py

```diff
@@ -17,12 +18,13 @@
 def add_spam_status_to_tagged_users(state, schema):
     OSFUser = state.get_model('osf', 'osfuser')
     users_with_tag = OSFUser.objects.filter(tags__name__in=TAG_MAP.keys()).prefetch_related('tags')
+    users_to_update = []
     for user in users_with_tag:
         for tag, value in TAG_MAP.items():
-            if user.tags.filter(name=tag).exists():
+            if user.tags.filter(system=True, name=tag).exists():
```

Comment:
Real quick, I think you need to to filter `system_tags`

I've added a regular tag "grapes" and a system tag "hello" to the user:
```shell
In [20]: u.tags.all()
Out[20]: <QuerySet [<Tag: grapes>]>

In [21]: u.tags.filter(system=True)
Out[21]: <QuerySet []>

In [22]: u.all_tags.all()
Out[22]: <QuerySet [<Tag: grapes>, <Tag: System Tag: hello>]>

In [23]: u.system_tags.all()
Out[23]: <QuerySet [u'hello']>
```

Reply:
ah I see! Didn't mean to butt in early.  

One thought, because you're importing OSFUser like this `OSFUser = state.get_model('osf', 'osfuser')`, I think these models are created dynamically, so not all of the mixin-related items are present.   I believe on the wiki migrations we had to get around this - when we were trying to load guids on NodeWikiPages.  Instead of pulling the guid off of NodeWikiPage, I had to load the Guid object directly.  You might experiment with loading the Tag object directly, or similar.

Reply:
Oh really really good idea, I'll try that!

Reply:
The mystery continues! I think loading `Tag` directly is still technically is a bit of a hand-waving thing though because `Tag` has 2 managers, `objects` with no system tags and `all_tags`. Accessing `all_tags` which should have them. In the migration, I get the same kind of error trying to use `all_tags`:
```
if Tag.all_tags.filter(osfuser=user, name=tag, system=True).exists():
AttributeError: type object 'Tag' has no attribute 'all_tags'
```
But doing this:
```
ipdb>  Tag.objects.get(osfuser=user, name=tag, system=True).name
u'ham_confirmed'
```
Gives me the tag I'm looking for.

What I am also really confused about tho is that accessing just `tags` in the migration also seems to pull in the system tags, even though I know it shouldn't in the "real world" because of the object manager.
```
ipdb> user.tags.filter(system=True).values_list('name', flat=True)
<QuerySet [u'ham_confirmed']>
```

This leaves me wondering if it's best to access `user.tags` knowing it's gonna have system tags because of how it's loaded in the migration, vs  loading the `Tag` object with also the wrong object manager `objects` vs `all_tags`. I am so confused on this!

Reply:
For clarity, seems indeed since the custom managers don't have `use_in_migrations = True`, they aren't being used here and we can just use `user.tags` after all! [Source](https://docs.djangoproject.com/en/2.0/topics/migrations/#historical-models)